backend/src/app/db/models/user.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+import datetime
+from typing import TYPE_CHECKING
+
+from advanced_alchemy.base import UUIDAuditBase
+from advanced_alchemy.types import DateTimeUTC
+from sqlalchemy import String
+from sqlalchemy.ext.hybrid import hybrid_property
+from sqlalchemy.orm import Mapped, mapped_column, relationship, validates
+
+if TYPE_CHECKING:
+    from .user_profile import UserProfile
+    from .user_role import UserRole
+
+
+class User(UUIDAuditBase):
+    __tablename__ = "user_account"
+    __table_args__ = {"comment": "User accounts for application access"}
+    __pii_columns__ = {"email"}
+
+    email: Mapped[str] = mapped_column(unique=True, index=True, nullable=False)
+    hashed_password: Mapped[str | None] = mapped_column(
+        String(length=255), nullable=True, default=None
+    )
+    magic_link_hashed_token: Mapped[str | None] = mapped_column(
+        String(length=255), nullable=True, default=None
+    )
+    magic_link_sent_at: Mapped[datetime.datetime] = mapped_column(
+        DateTimeUTC(timezone=True), default=None, nullable=True
+    )
+
+    @validates("magic_link_sent_at")
+    def validate_tz_info(self, _: str, value: datetime.datetime) -> datetime.datetime:
+        if value and value.tzinfo is None:
+            value = value.replace(tzinfo=datetime.timezone.utc)
+        return value
+
+    terms_accepted: Mapped[bool] = mapped_column(default=False, nullable=False)
+    is_active: Mapped[bool] = mapped_column(default=False, nullable=False)
+    is_superuser: Mapped[bool] = mapped_column(default=False, nullable=False)
+    is_verified: Mapped[bool] = mapped_column(default=False, nullable=False)
+    verified_at: Mapped[datetime.date] = mapped_column(nullable=True, default=None)
+    joined_at: Mapped[datetime.date] = mapped_column(default=datetime.datetime.now)
+    login_count: Mapped[int] = mapped_column(default=0)
```

Comment:
Nit: Do we really need this? Also, where/when will this be incremented? Wouldn't we want an authentication journal instead (later on)?

Reply:
Bad copy paste indeed :+1: